tools/genesis-snapshot/presets/presets_yaml.go

```diff
@@ -0,0 +1,108 @@
+package presets
+
+import (
+	"fmt"
+	"os"
+
+	"golang.org/x/crypto/blake2b"
+	"gopkg.in/yaml.v3"
+
+	"github.com/iotaledger/hive.go/crypto/ed25519"
+	"github.com/iotaledger/hive.go/lo"
+	"github.com/iotaledger/hive.go/runtime/options"
+	"github.com/iotaledger/iota-core/pkg/testsuite/mock"
+	"github.com/iotaledger/iota-core/pkg/testsuite/snapshotcreator"
+	iotago "github.com/iotaledger/iota.go/v4"
+	"github.com/iotaledger/iota.go/v4/hexutil"
+)
+
+type ValidatorYaml struct {
+	Name      string `yaml:"name"`
+	PublicKey string `yaml:"public-key"`
+}
+
+type BlockIssuerYaml struct {
+	Name      string `yaml:"name"`
+	PublicKey string `yaml:"public-key"`
+}
+
+type BasicOutputYaml struct {
+	Address string `yaml:"address"`
+	Amount  uint64 `yaml:"amount"`
+	Mana    uint64 `yaml:"mana"`
+}
+
+type ConfigYaml struct {
+	Name     string `yaml:"name"`
+	FilePath string `yaml:"filepath"`
+
+	Validators   []ValidatorYaml   `yaml:"validators"`
+	BlockIssuers []BlockIssuerYaml `yaml:"block-issuers"`
+	BasicOutputs []BasicOutputYaml `yaml:"basic-outputs"`
+}
+
+func GenerateFromYaml(hostsFile string) ([]options.Option[snapshotcreator.Options], error) {
+	yamlFile, err := os.ReadFile(hostsFile)
+	if err != nil {
+		return nil, err
+	}
+	var configYaml ConfigYaml
+	err = yaml.Unmarshal(yamlFile, &configYaml)
+	if err != nil {
+		return nil, err
+	}
+
+	var accounts []snapshotcreator.AccountDetails
+	for _, validator := range configYaml.Validators {
+		pubkey := validator.PublicKey
+		fmt.Printf("adding validator %s with publicKey %s\n", validator.Name, pubkey)
+		account := snapshotcreator.AccountDetails{
+			AccountID:            blake2b.Sum256(lo.PanicOnErr(hexutil.DecodeHex(pubkey))),
+			Address:              iotago.Ed25519AddressFromPubKey(lo.PanicOnErr(hexutil.DecodeHex(pubkey))),
+			Amount:               mock.MinValidatorAccountAmount(protocolParamsDocker),
+			IssuerKey:            iotago.Ed25519PublicKeyBlockIssuerKeyFromPublicKey(ed25519.PublicKey(lo.PanicOnErr(hexutil.DecodeHex(pubkey)))),
+			ExpirySlot:           iotago.MaxSlotIndex,
+			BlockIssuanceCredits: iotago.MaxBlockIssuanceCredits / 4,
+			StakingEndEpoch:      iotago.MaxEpochIndex,
+			FixedCost:            1,
```

Comment:
We could make it more flexible and allow setting all fields. But if it's not needed now then it's fine to always go with the defaults.

Reply:
This was my 2nd attempt - the first was to have a json file with that can overwrite every setting of the default parameters from iota.go ... but it was too complicated actually and not needed at all. I just needed a way to generate a snapshot based on my ansible deployment.

Maybe we can extend it later if we need something additionally but I have the feeling pimping it up now wouldn't give much advantage.